### PR TITLE
Update add_metric() docstring for Keras 3 compatibility

### LINKED ISSUE #110884: Documentation mismatch: Layer/Model.add_metric() shown usable in call(), but Keras 3 disables / breaks it (TF 2.17 + Keras 3.4)
### Issue type

Documentation Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

source

### TensorFlow version

2.17.0

### Custom code

Yes

### OS platform and distribution

Ubuntu 24.04.3 LTS

### Mobile device

_No response_

### Python version

3.10.19

### Bazel version

_No response_

### GCC/compiler version

_No response_

### CUDA/cuDNN version

CUDA: 12.3, cuDNN: 8

### GPU model and memory

4 × NVIDIA RTX 3090 (24GB each)

### Current behavior?

- TensorFlow guide(s) state that layers/models can track metrics via `add_metric()` (often used inside `call()` in subclassed models).
- On TF 2.17.0 + Keras 3.4.1 (TF backend), using `self.add_metric(...)` inside `call()` is not a reliable supported pattern. This aligns with existing reports where `add_metric` is “permanently disabled” in the codepath.
- As a result, the docs example/messaging is misleading for Keras 3 users.

**Expected behavior**
- TensorFlow docs should be updated to not recommend `add_metric()` inside `call()` for Keras 3.
- Docs should show the Keras 3-compatible approach (create a `keras.metrics.Metric` object in `__init__`, call update_state, expose via metrics property, or use a custom `train_step`).

### Standalone code to reproduce the issue

```shell
import os
os.environ["KERAS_BACKEND"] = "tensorflow"

import numpy as np
import tensorflow as tf
import keras

print("TF:", tf.__version__)
print("Keras:", keras.__version__)
print("Backend:", keras.backend.backend())

class M(keras.Model):
    def __init__(self):
        super().__init__()
        self.d = keras.layers.Dense(4)

    def call(self, x):
        y = self.d(x)
        m = keras.losses.MeanSquaredError()(x, y)
        # This is the pattern implied by TF guides ("track metrics via add_metric")
        self.add_metric(m, name="reconstruction_loss")
        return y

x = np.random.randn(8, 4).astype("float32")
m = M()

# Force eager + disable jit to avoid GPU/XLA changing warning surfaces
m.compile(optimizer="adam", loss="mse", run_eagerly=True, jit_compile=False)

try:
    hist = m.fit(x, x, epochs=1, verbose=0)
    print("history keys:", sorted(list(hist.history.keys())))
except Exception as e:
    print("EXCEPTION:", type(e).__name__, str(e))

print("metrics:", [mm.name for mm in m.metrics])
```

### Relevant log output

```shell
2026-02-23 00:24:36.838904: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:485] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2026-02-23 00:24:36.859605: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:8454] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2026-02-23 00:24:36.865974: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1452] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
```

### PR DESCRIPTION
**Pull request by mvanhorn**
## Summary

Fixes #110884

The `add_metric()` docstring recommends calling it inside `call()` for subclassed layers/models. However, this pattern is **disabled in Keras 3** (default since TF 2.16+), causing confusion for users following the documented examples.

### Changes

Added a `Note` at the top of the `add_metric()` docstring in `base_layer.py` that:
- Warns that `add_metric()` inside `call()` is not supported in Keras 3
- Points users to the Keras 3-compatible approach:
  - Create `keras.metrics.Metric` objects in `__init__()`
  - Call `update_state()` during forward pass
  - Expose metrics via the `metrics` property
- Clarifies the existing example is for legacy `tf.keras` (Keras 2)

## Test plan

- [x] Pure documentation change - no code logic modified
- [x] Existing tests unaffected

This contribution was developed with AI assistance (Claude Code).

### CODE REVIEW

**Comment by mvanhorn** [comment]
@googlebot I signed it!

**Comment by mvanhorn** [comment]
I have signed the Google CLA.

**Comment by keerthanakadiri** [comment]
Hi @rohan100jain, Can you kindly review this PR? Thank you!

**Comment by mvanhorn** [comment]
Thanks for the ping, @keerthanakadiri! Happy to address any feedback if needed.

**Comment by mvanhorn** [comment]
Closing this — I should have checked the folder before opening it. `tensorflow/python/keras/` carries a README that starts "STOP! This folder contains the legacy Keras code which is stale and about to be deleted", so a docstring correction here isn't worth reviewer time even if the wording is off.

Apologies for the noise.
